src/ChangeStream/ResourceStack.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository\ChangeStream;
+
+use Puli\Repository\Api\Resource\PuliResource;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+use RuntimeException;
+use Webmozart\Assert\Assert;
+
+/**
+ * Represents a versionned stack of resources for a given path.
+ *
+ * You can access different versions of a resource using a ChangeStream
+ * that will return you resource stacks.
+ *
+ * @since  1.0
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class ResourceStack extends ArrayResourceCollection
```

Comment:
What do you mean? The built resource stack does contains only resources objects and they are different even if they are versions of the same path.